Dockerfile

```diff
@@ -1,20 +1,17 @@
-FROM golang:1.16.10-buster
+FROM debian:bookworm
 EXPOSE 8000
 
-ENV GODEBUG=x509ignoreCN=0
```

Comment:
Hm, this should still be required to work correctly. Nervy that it does. Might be making leaf certificates incorrectly? Did I miss something thought?

Reply:
To explain this one a little bit, the `x509ignoreCN` environment variable tells golang to weaken its certificate subject name checks and accepts certificates which only contain a `CN` field when trying to match a DNS name. Support for this variable was removed in Go 1.17 (see notes for crypto/x509 [here](https://go.dev/doc/go1.17#minor_library_changes)). And a valid certificate, as far as go is concerned, must now use the X509v3 `subjectAltName` extension to provide a DNS name.

This caused a problem for the integration tests, because the test certificate for the DB only provides `CN=db` and this is no longer sufficient for go 1.17, which is the reason for commit 43fb9573f79a900744e0786b1af00acb52854d35 in this PR to update the test certs.

Reply:
Ah, I was worried that variable applied at certificate creation, not just verification, time and we'd be generate certs without a CommonName and only a SAN but that's not the case (from looking up the older docs for [VerifyHostname](https://pkg.go.dev/crypto/x509@go1.16.15#Certificate.VerifyHostname) and writing [a test impl](https://gist.github.com/jmhodges/42c8b1cb14451844014b00dc16a45a31)). 

I think I saw the cose_test.go and verifier_test.go changes thought they might be related. And I'm extra paranoid about this because this month we had [to correct a bug](https://github.com/mozilla-services/autograph/pull/884) around us making the CN).

Any idea why the error messages for cose_test.go and verifier_test.go got worse? Not a strong request but maybe you know already

Reply:
The changes in the tests are due to some refactoring of the Go certificate parsing routines in `crypto/x509` that changed the error messages that library generates when we feed it some invalid ASN.1. From the Go 1.17 [changelog](https://go.dev/doc/go1.17#minor_library_changes):
> [ParseCertificate](https://go.dev/pkg/crypto/x509/#ParseCertificate) has been rewritten, and now consumes ~70% fewer resources. The observable behavior when processing WebPKI certificates has not otherwise changed, except for error messages.

Since our unit tests are using the error messages as the pass/fail conditions, they need to be updated to accommodate new versions of Go when those messages change.

Reply:
Hell yeah, thank you